fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimer.java

```diff
@@ -74,6 +76,66 @@ public static void checkEventNameUniqueness(@Nonnull Stream<Event> events) {
         }
     }
 
+    /**
+     * Subtracts counts and times recorded by a snapshot of a timer.
+     *
+     * Subtracting a snapshot of a timer from the original timer after subsequent operations have
+     * been performed and timed can provide useful metrics as to the cost of those subsequent operations.
+     *
+     * The timer must have been previously derived from the snapshot. Moreover,
+     * the timer should not have been reset after the snapshot was taken.
+     *
+     * @param timer timer to subtract the snapshot from
+     * @param timerSnapshot snapshot of the provided timer
+     *
+     * @return a snapshot of the provided timer
+     */
+    @Nonnull
+    public static StoreTimer subtractSnapshot(@Nonnull StoreTimer timer, @Nonnull StoreTimerSnapshot timerSnapshot) {
+
+        if (timer.hashCode() != timerSnapshot.getFromID()) {
+            throw new UnsupportedOperationException("Invalid to subtract a snapshot timer from a timer it was not derived from.");
+        }
+
+        if (timer.lastReset > timerSnapshot.getCreateTime()) {
+            throw new UnsupportedOperationException("Invalid to substract a snapshot timer from a timer that has been reset after the snapshot was taken");
+        }
+
+        StoreTimer resultTimer = new StoreTimer();
+        timer.counters.entrySet()
+                .stream()
+                .forEach(entry -> {
+                    Counter diffCounter = new Counter();
+                    int count = 0;
+                    long time = 0L;
+                    if (timerSnapshot.containsCounter(entry.getKey())) {
+                        count = timerSnapshot.getCounterSnapshot(entry.getKey()).getCount();
+                        time = timerSnapshot.getCounterSnapshot(entry.getKey()).getTimeNanos();
+                    }
+                    diffCounter.count.set(entry.getValue().count.get() - count);
+                    diffCounter.timeNanos.set(entry.getValue().timeNanos.get() - time);
+                    resultTimer.counters.put(entry.getKey(), diffCounter);
+                });
+        timer.timeoutCounters.entrySet()
+                .stream()
+                .forEach(entry -> {
+                    Counter diffCounter = new Counter();
+                    int count = 0;
+                    long time = 0L;
+                    if (timerSnapshot.containsTimeoutCounter(entry.getKey())) {
+                        count = timerSnapshot.getTimeoutCounterSnapshot(entry.getKey()).getCount();
+                        time = timerSnapshot.getTimeoutCounterSnapshot(entry.getKey()).getTimeNanos();
+                    }
+                    diffCounter.count.set(entry.getValue().count.get() - count);
+                    diffCounter.timeNanos.set(entry.getValue().timeNanos.get() - time);
+                    resultTimer.timeoutCounters.put(entry.getKey(), diffCounter);
+                });
```

Comment:
These two are the same logic and should probably become: 

    subtract(timer.counters, timerSnapshot, resultTimer);
    subtract(timer.timeoutCounters, timerSnapshot, resultTimer);

and then move the common logic into `subtract`

Reply:
Yeah can refactor that a bit.